data-prepper-api/src/main/java/org/opensearch/dataprepper/model/trace/JacksonSpan.java

```diff
@@ -196,6 +205,31 @@ public Builder getThis() {
             return this;
         }
 
+        /**
+         * Sets the data of the event.
+         * @param data JSON representation of the data
+         * @since 2.0
+         */
+        public Builder withJsonData(final String data) {
+            try {
+                this.data.putAll(mapper.readValue(data, MAP_TYPE_REFERENCE));
+            } catch (final JsonProcessingException e) {
+                throw new RuntimeException(String.format("An exception occurred due to invalid JSON while reading event data: %s", data), e);
```

Comment:
Is this customer data in the exception ? How are we handling serialization/deserialization errors in other part of the repository ?

Reply:
This is the data sent by core peer forwarder which is customer data. Should we remove the data from exception message?
Even I was confused how this is handled in the data-prepper-api. This is how it's handled in JacksonEvent class  https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java#L101
```
try {
                return mapper.readTree((String) data);
            } catch (final JsonProcessingException e) {
                throw new IllegalArgumentException("Unable to convert data into an event");
            }
```

Reply:
yes, lets remove data from exception.